src/prefect/flows.py

```diff
@@ -158,6 +159,25 @@ def __init__(
             List[Callable[[schemas.core.Flow, FlowRun, State], None]]
         ] = None,
     ):
+        # Validate if hooks type is list and hook_type is callable
+        hooks = [on_completion, on_failure, on_cancellation, on_crashed]
+        for hook_types in hooks:
```

Comment:
Should probably swap use of `hooks` and `hook_types` here for clarity.

Reply:
Sure thing, the names are now confusing me too.